src/keria/app/agenting.py

```diff
@@ -326,63 +201,95 @@ def __init__(self, name, bran, base="", releaseTimeout=None,
         self.iurls = iurls
         self.durls = durls
 
-        if self.configFile is not None:
+        if cf is None and self.configFile is not None:
             self.cf = configing.Configer(name=self.configFile,
                                          base="",
                                          headDirPath=self.configDir,
-                                         temp=False,
+                                         temp=temp,
                                          reopen=True,
                                          clear=False)
+        else:
+            self.cf = cf
 
         self.agents = dict()
 
         self.adb = adb if adb is not None else basing.AgencyBaser(name="TheAgency", base=base, reopen=True, temp=temp)
         super(Agency, self).__init__(doers=[Releaser(self, releaseTimeout=releaseTimeout)])
 
-    def create(self, caid, salt=None):
+    def _load_config_for_agent(self, caid):
         """
-        Create and return a new agent with the given caid and optional salt.
-
-        Returns:
-            Agent: The newly created agent.
+        Loads configuration data for an agent by looking up the Agency's configuration and copying
+        the agency config for the agent merged with curls, iurls, and durls specified by environment
+        variables.
 
         Parameters:
-            caid (str): The controller AID (Agent Identifier) for the new agent.
-            salt (str): Optional QB64 salt for the agent's Habery. If not provided, a random salt will be used.
+            caid (str): The controller AID (Agent Identifier) for the agent.
+        Returns:
+            dict: A dictionary containing the agent's configuration data.
         """
-        ks = keeping.Keeper(name=caid,
-                            base=self.base,
-                            temp=self.temp,
-                            reopen=True)
-
         timestamp = nowIso8601()
-        data = dict(self.cf.get() if self.cf is not None else { "dt": timestamp })
```

Comment:
I moved the configuration reading and writing to separate functions in order to clean up this agent create function.